### PR TITLE
feat: Update to new github action version

### PR DESCRIPTION
**Pull request by blopezpi**
Improvements for the new github action.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -39,28 +38,43 @@ jobs:
...
+      - if: ${{ !inputs.component_app }}
+        name: Generate a release tag
+        uses: intelygenz/monorepo-tagger-action@v1.2.1
+        id: release_tag_component
         with:
           mode: 'component'
           type: 'final'
           component-prefix: "${{ inputs.component }}-"
+          strip-component-from-tag: true
+          update-versions-in: '[{"file": "${{ inputs.values_file }}", "property": "${{ inputs.component_helm }}"}]'
```
Comment (alejandrogr) [inline .github/workflows/build.yaml:56]: What do you think of renaming the property `component_helm` to `helm_component_version` or `helm_component_tag`?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -70,18 +84,35 @@ jobs:
...
-          TAG=`yq r helm/${{ inputs.chart_name }}/values.yaml '${{ inputs.component_helm }}.image.tag'`
+          TAG=`yq r ${{ inputs.values_file }} '${{ inputs.component_helm }}'`
           echo ::set-output name=TAG::${TAG}
           echo Component version in release branch is ${TAG}
 
-      - name: Generate a fix
-        uses: intelygenz/monorepo-tagger-action@v1.0
-        id: fix_tag
+      - if: ${{ !inputs.component_app }}
+        name: Generate a fix
```
Comment (alejandrogr) [inline .github/workflows/build.yaml:92]: Name should be: `Generate a fix tag for component`

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -117,7 +146,6 @@ jobs:
         id: set_tag
```
Comment (alejandrogr) [inline .github/workflows/build.yaml:146]: Do we need this step at all?

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -117,7 +146,6 @@ jobs:
...
         run: |
           VERSION=${{needs.create-release-tag.outputs.tag}}${{needs.create-fix-tag.outputs.tag}}
-          VERSION=`echo $VERSION | sed "s/${{ inputs.component }}-//"`
           echo Tag set to ${VERSION}
           echo ::set-output name=VERSION::${VERSION}
```
Comment (alejandrogr) [inline .github/workflows/build.yaml:151]: In the bellow step with name `Push to Docker Hub` we are using `${{ steps.version.outputs.tag }}` to tag the docker image, but I don't see a `version` step with a `tag` output. where does this variable come from?

**Review by alejandrogr — COMMENTED** [review]
This is a very nice job! I like the templating approach to simplify the workflows in the repositories.

Reviewing the PR I starting thinking that the default value for the action input `strip-component-from-tag` should be `true` instead of `false`. Almost in every situation I can think off you want to strip the `component-prefix` from the tag to use it for whatever reason in another steps or actions. What do you think?

**Comment by alejandrogr** [comment]
I have another question:

In the step with name `Push to Docker Hub` we are using `${{ steps.version.outputs.tag }}` to tag the docker image, but I don't see a `version` step with a `tag` output. where does this variable come from?

File: .github/workflows/build.yaml (hunk 3 above)
Comment (blopezpi) [reply .github/workflows/build.yaml:146]: Yes, we need this because the tag in some cases comes from the `create-fix-tag` or the `create-release-tag`.

**Comment by blopezpi** [comment]
> I have another question:
> 
> In the step with name `Push to Docker Hub` (line 152 in `.github/workflows/build.yaml `) we are using `${{ steps.version.outputs.tag }}` to tag the docker image, but I don't see a `version` step with a `tag` output. where does this variable come from?

The step that we are using to get the tag is the `set_tag` step that catch the output for the fix or create jobs and set an output with the finally tag version.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (blopezpi) [reply .github/workflows/build.yaml:56]: We can change it.

**Comment by blopezpi** [comment]
> This is a very nice job! I like the templating approach to simplify the workflows in the repositories.
> 
> Reviewing the PR I starting thinking that the default value for the action input `strip-component-from-tag` should be `true` instead of `false`. Almost in every situation I can think off you want to strip the `component-prefix` from the tag to use it for whatever reason in another steps or actions. What do you think?

I think that we need to strip the component prefix in all ours components, so we can change the default value to `true`.

**Review by aacecandev — APPROVED** [review]

**Review by alejandrogr — APPROVED** [review]

**Pull request merged**